src/inferenceql/query/scalar.cljc

```diff
@@ -112,23 +113,26 @@
                        :env m}))
       result)))
 
-(defn prob
-  [model event]
-  (let [event (inference-event event)]
-    (math/exp (gpm/logprob model event))))
-
-(defn pdf
-  [model event]
-  (let [event (update-keys event str)]
-    (math/exp (gpm/logpdf model event {}))))
-
-(defn condition
-  [model conditions]
-  (let [conditions (-> (medley/filter-vals some? conditions)
-                       (update-keys str))]
-    (cond-> model
-      (seq conditions)
-      (gpm/condition conditions))))
+(def prob (cache/lru
+            (fn prob*
+              [model event]
+              (let [event (inference-event event)]
+                (math/exp (gpm/logprob model event))))))
+
+(def pdf (cache/lru
+           (fn pdf*
+             [model event]
+             (let [event (update-keys event str)]
+               (math/exp (gpm/logpdf model event {}))))))
+
+(def condition (cache/lru
```

Comment:
tiny point and entirely a matter of taste: I'd be sad at how much horizontal real estate the def part cost me and would put a newline in front of `(cache/lru` and see if it looked better.

Reply:
Done